index.d.ts

```diff
@@ -0,0 +1,60 @@
+export interface Action {
+  type: string;
+}
+
+export type Reducer<S> = (state: S, action: Action) => S;
+
+export type Dispatch = (action: any) => any;
```

Comment:
We can't avoid `any` here. Middleware potentially allows `dispatch` to accept anything: functions (thunks), promises, etc. Its return type could also be anything.